app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt

```diff
@@ -24,88 +21,45 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         mBinding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(mBinding.root)
-        showDate()
-        showCalendar()
     }
 
-    private fun initCalendar() {
-        val c = Calendar.getInstance()
-        val year = c.get(Calendar.YEAR)
-        val month = c.get(Calendar.MONTH)
-        val day = c.get(Calendar.DAY_OF_MONTH)
-
-
-
-
-        val dpd = DatePickerDialog(this, DatePickerDialog.OnDateSetListener { view, year, monthOfYear, dayOfMonth ->
+    override fun onStart() {
+        super.onStart()
+        mBinding.viewModel = viewModel
+        mBinding.lifecycleOwner = this
 
-            var monthString = (monthOfYear+1).toString();
-            if (monthString.length == 1) {
-                monthString = "0$monthString"
-            }
-            mDate = "$year-$monthString-${dayOfMonth-1}"
-            loadData()
+        initListener()
+        initObserver()
+        displayFormatDate()
 
-        }, year, month, day)
-        var dt = Date()
-        dt = c.time
-        dpd.datePicker.maxDate = dt.time
-        dpd.show()
+        viewModel.getStatusCovid()
     }
 
-    private fun loadData() {
-        viewModel.getStatusCovid(mDate)
-        viewModel.statusCovidLivedata.observe(this) {
-            when (it) {
-                is Resource.Success -> {
-                    hideLoading()
-                    mBinding.tvConfirmed.text =  "${getString(R.string.confirmed_cases)} ${it.data.confirmed}"
-                    mBinding.tvDeaths.text = "${getString(R.string.number_of_deceased_people)} ${it.data.deaths}"
-                }
-                is Resource.Loading -> {
-                    showLoading()
-                }
-                is Resource.Error -> {
-                    hideLoading()
-                    it.message?.let {
-                        Toast.makeText(this, "An error ocurred : $it", Toast.LENGTH_LONG).show()
-                    }
-                }
-                else -> {
+    private fun initCalendar() {
+        CalendarManager.showPickerDialog(this) {
+            mDate = it
+            displayFormatDate(it)
+            viewModel.getStatusCovid(mDate.toDateFormat(YYYY_MM_DD_WITH_SCRIPT))
+        }
+    }
 
+    private fun initObserver() {
+        viewModel.statusCovidStateLiveData.observe(this) {
+            if (it is State.Error) {
+                mBinding.errorView.show("An error ocurred : ${it.message}") {
```

Comment:
In this case,  it is better to use String.xml for any text.

Reply:
String resource is created.